exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillMergeProjectRule.java

```diff
@@ -166,4 +169,25 @@ public void onMatch(RelOptRuleCall call) {
     return list;
   }
 
+  public static Project replace(Project topProject, Project bottomProject) {
```

Comment:
Can `ProjectAllowDup` and `Project` rels be merged?
For instance:
`00-01      ProjectAllowDup(MONTHID=[$0], NOFOCONTRACTS=[$1]) : rowType = RecordType(INTEGER MONTHID, INTEGER NOFOCONTRACTS)`
`00-02        Project(MonthId=[CAST($1):INTEGER], No of Contracts=[CAST($2):INTEGER]) : rowType = RecordType(INTEGER MonthId, INTEGER No of Contracts)`

Reply:
They can be merged into the top level Project that allows duplicates...however, the purpose of the `replace()` method here is to simply allow the caller to replace a project with another with the assumption that callers know exactly what they are doing.  This is not applying the full fledged `DrillMergeProjectRule`.

Reply:
It makes sense. A small javadoc can be useful here.

Reply:
Added javadoc.